worker/src/RTC/Producer.cpp

```diff
@@ -605,6 +607,56 @@ namespace RTC
 		}
 	}
 
+	void Producer::HandleNotification(PayloadChannel::Notification* notification)
+	{
+		MS_TRACE();
+
+		switch (notification->eventId)
+		{
+			case PayloadChannel::Notification::EventId::PRODUCER_SEND:
+			{
+				const auto* data = notification->payload;
+				auto len         = notification->payloadLen;
+
+				// Increase receive transmission.
+				this->listener->OnProducerReceiveData(this, len);
+
+				if (len > RTC::MtuSize + 100)
+				{
+					MS_WARN_TAG(rtp, "given RTP packet exceeds maximum size [len:%zu]", len);
+
+					return;
+				}
+
+				// If this is the first time to receive a RTP packet then allocate the receiving buffer now.
+				if (!this->buffer)
+					this->buffer = new uint8_t[RTC::MtuSize + 100];
+
+				// Copy the received packet into this buffer so it can be expanded later.
+				std::memcpy(this->buffer, data, static_cast<size_t>(len));
+
+				RTC::RtpPacket* packet = RTC::RtpPacket::Parse(this->buffer, len);
+
+				if (!packet)
+				{
+					MS_WARN_TAG(rtp, "received data is not a valid RTP packet");
+
+					return;
```

Comment:
break

Reply:
@nazar-pc, the `buffer` belonged, prior to this PR, to DirectTransport, hence all Producers would use the same. Within this PR every Producer will own one, with the corresponding memory usage increase if many Producers in the same DirectTransport need it. In fact it could be a static buffer that is used by every instance. cc @ibc.

Reply:
Agree, applied corresponding change in latest commit